apps/web/src/ee/billing/components/ActivePlanBanner.tsx

```diff
@@ -1,36 +1,40 @@
 import { Text, Title } from '@novu/novui';
 import { MantineTheme } from '@mantine/core';
 import { css } from '@novu/novui/css';
+import { da } from 'date-fns/locale';
 import { UsageProgress } from './UsageProgress';
-import { useSubscriptionContext } from './SubscriptionProvider';
+import { useSubscriptionContext, type UseSubscriptionType } from './SubscriptionProvider';
 import { capitalizeFirstLetter } from '../../../utils/string';
 import { Badge } from './Badge';
 import { PlanActionButton } from './PlanActionButton';
 
-export const ActivePlanBanner = ({ selectedBillingInterval }: { selectedBillingInterval: 'month' | 'year' }) => {
-  const { apiServiceLevel, status, events, trial } = useSubscriptionContext();
+type BillingInterval = 'month' | 'year';
+
+export const ActivePlanBanner = ({ selectedBillingInterval }: { selectedBillingInterval: BillingInterval }) => {
+  const subscription = useSubscriptionContext();
 
   return (
     <div className={styles.activePlanWrapper}>
       <Title variant="section">Active Plan</Title>
       <div className={styles.banner}>
         <div className={styles.content}>
-          <PlanHeader apiServiceLevel={apiServiceLevel} isFreeTrialActive={trial.isActive} daysLeft={trial.daysLeft} />
-          <PlanInfo apiServiceLevel={apiServiceLevel} currentEvents={events.current} maxEvents={events.included} />
+          <PlanHeader {...subscription} />
+          <PlanInfo {...subscription} />
         </div>
-        <PlanActions trialEnd={trial.end} status={status} selectedBillingInterval={selectedBillingInterval} />
+        <PlanActions {...subscription} selectedBillingInterval={selectedBillingInterval} />
       </div>
     </div>
   );
 };
 
-const PlanHeader = ({ apiServiceLevel, isFreeTrialActive, daysLeft }) => {
+function PlanHeader({ apiServiceLevel, trial }: UseSubscriptionType) {
```

Comment:
I always prefer simple function components to avoid the unnecessary binding of this by using arrow functions.